runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -6452,6 +6452,71 @@ class INTERPRETER_CLASS
 		return rc;
 	}
 
+	static void
+	referenceFixOSlotIterator(J9VMThread *vmThread, J9StackWalkState *walkState, j9object_t *oSlotPointer, const void * stackLocation)
+	{
+		if (*oSlotPointer == (j9object_t)walkState->userData1) {
+			*oSlotPointer = (j9object_t)walkState->userData2;
+		}
+	}
+
+
+	VMINLINE void
+	heapifyObjectIfRequired(REGISTER_ARGS_LIST, j9object_t original)
+	{
+		if ((UDATA)original < (UDATA)(_currentThread->stackObject) || (UDATA)original >= (UDATA)(_currentThread->stackObject->end))
+			return;
+		j9object_t copy = NULL;
+		J9Class *objectClass = J9OBJECT_CLAZZ(_currentThread, original);
+		UDATA flags = J9CLASS_FLAGS(objectClass);
+		if (flags & J9AccClassArray) {
+			U_32 size = J9INDEXABLEOBJECT_SIZE(_currentThread, original);
+			copy = VM_VMHelpers::inlineAllocateIndexableObject(_currentThread, &_objectAllocate, objectClass, size, false, false, false);
+
+			if (NULL == copy) {
+				updateVMStruct(REGISTER_ARGS);
+				copy = _vm->memoryManagerFunctions->J9AllocateIndexableObject(_currentThread, objectClass, size, J9_GC_ALLOCATE_OBJECT_NON_INSTRUMENTABLE);
+				VMStructHasBeenUpdated(REGISTER_ARGS);
+				if (J9_UNEXPECTED(NULL == copy)) {
+					/* TODO: Add exception logic here */
```

Comment:
Not sure how to set exceptions here since this doesn't correspond to a BytecodeAction, and would probably depend on how it is called.